### PR TITLE
Calculate and display Benchmark Group summaries, using Geometric Means

### PR DESCRIPTION
**Pull request by terceiro**

### CODE REVIEW

**Comment by terceiro** [comment]
screenshots:

geomeans-overall

geomeans-select

File: benchmarks/models.py [hunk 1]
```diff
@@ -290,6 +292,40 @@ def __unicode__(self):
...
+        # multiply the n values, then take the nth-root of the result) does not
+        # work in practice. When you multiple an large enough amount of large
+        # enough numbers, their product will oferflow the float representation,
+        # and the result will be Infinity.
+        #
+        # Will use the alternative method described in
+        # https://en.wikipedia.org/wiki/Geometric_mean -- topic "Relationship
+        # with arithmetic mean of logarithms" -- which is exp(sum(log(x_i)/n))
+        n = len(values)
+        log_sum = reduce(lambda x,y: x + y, map(log, values))
```
Comment (mwasilew) [inline benchmarks/models.py]: I guess we need to check if value list doesn't contain '0'.

File: benchmarks/tasks.py [hunk 2]
```diff
@@ -74,6 +77,21 @@ def store_testjob_data(testjob, test_results):
                 test_job_id=testjob.id,
                 benchmark=benchmark
             )
+            if benchmark_group:
+                for v in values:
```
Comment (mwasilew) [inline benchmarks/tasks.py]: Alternatively check here if v != 0 to avoid log(0) in BenchmarkSummary object.

**Comment by mwasilew** [comment]
Looks almost OK. Should I merge now and patch for geomean calculation later (geomean of series including 0)?

**Comment by terceiro** [comment]
yeah I think we can fix that later

**Pull request merged**